### PR TITLE
12889_conv_round_corners

### PR DESCRIPTION
**Pull request by dominicmarmionswrve**
Rounded corners on conversations

https://swrvedev.jira.com/browse/SWRVE-12889

@Sergio-Mira please

### CODE REVIEW

**Comment by dominicmarmionswrve** [comment]
retest this please

File: SwrveConversationSDK/src/main/java/com/swrve/sdk/SwrveConversationHelper.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import android.graphics.Paint;
+import android.graphics.RectF;
+import android.graphics.drawable.Drawable;
+import android.graphics.drawable.LayerDrawable;
+import android.graphics.drawable.ShapeDrawable;
+import android.graphics.drawable.shapes.RoundRectShape;
+
+import com.swrve.sdk.conversations.R;
+
+public class SwrveConversationHelper {
```
Comment (Sergio-Mira) [inline SwrveConversationSDK/src/main/java/com/swrve/sdk/SwrveConversationHelper.java]: Could you add a comment saying this is internal?

Comment (dominicmarmionswrve) [inline SwrveConversationSDK/src/main/java/com/swrve/sdk/SwrveConversationHelper.java]: Good point. I'll add that comment, but I'll also move it to com.swrve.sdk.conversations.ui and make it package access only. :)

File: SwrveConversationSDK/src/main/java/com/swrve/sdk/conversations/ui/ConversationFragment.java [hunk 2]
```diff
@@ -219,17 +225,23 @@ private void renderControls(Activity activity) {
...
             if (content instanceof Content) {
                 Content modelContent = (Content) content;
                 String modelType = modelContent.getType().toString();
                 if (modelType.equalsIgnoreCase(ConversationAtom.TYPE_CONTENT_IMAGE)) {
-                    ConversationImageView iv = new ConversationImageView(activity, modelContent);
                     String filePath = swrveConversation.getCacheDir().getAbsolutePath() + "/" + modelContent.getValue();
                     if(SwrveHelper.hasFileAccess(filePath)) {
+                        ConversationImageView iv;
+                        if (i == 0 && pageBorderRadius > 0) { // only apply rounded corner if its the first atom in page
+                            iv = new ConversationImageViewRounded(activity, modelContent, pageBorderRadius, pageBgColor);
```
Comment (Sergio-Mira) [inline SwrveConversationSDK/src/main/java/com/swrve/sdk/conversations/ui/ConversationFragment.java]: Is this to crop borders on images?

Comment (Sergio-Mira) [inline SwrveConversationSDK/src/main/java/com/swrve/sdk/conversations/ui/ConversationFragment.java]: Could it not be done for the whole layout that contains the atoms?

Comment (dominicmarmionswrve) [inline SwrveConversationSDK/src/main/java/com/swrve/sdk/conversations/ui/ConversationFragment.java]: Yeah, need to clip corners on video and images. Doing it on the whole layout would be better but I couldn't get it to work :(

**Comment by dominicmarmionswrve** [comment]
retest this please

File: SwrveConversationSDK/src/main/java/com/swrve/sdk/conversations/ui/ConversationRoundedLinearLayout.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    public ConversationRoundedLinearLayout(Context context, AttributeSet attrs) {
+        super(context, attrs);
+    }
+
+    @Override
+    protected void onSizeChanged(int w, int h, int oldw, int oldh) {
+        super.onSizeChanged(w, h, oldw, oldh);
+        path.reset();
+        rect.set(0, 0, w, h);
+        int maxModalWidthPx = getResources().getDimensionPixelSize(R.dimen.swrve__conversation_max_modal_width);
```
Comment (Sergio-Mira) [inline SwrveConversationSDK/src/main/java/com/swrve/sdk/conversations/ui/ConversationRoundedLinearLayout.java]: ~ ~ Could you move this to the constructor? That way we only read it once

File: SwrveConversationSDK/src/main/java/com/swrve/sdk/conversations/ui/ConversationFragment.java [hunk 4]
```diff
@@ -248,7 +251,7 @@ private void renderContent(Activity activity) {
                     setBackgroundDrawable(view, colorStyle.getPrimaryDrawable());
                     contentLayout.addView(view);
                 } else if (modelType.equalsIgnoreCase(ConversationAtom.TYPE_CONTENT_VIDEO)) {
-                    YoutubeVideoView view = new YoutubeVideoView(activity, modelContent, fullScreenFrame);
+                    YoutubeVideoView view = new YoutubeVideoView(activity, modelContent, fullScreenFrame);;
```
Comment (Sergio-Mira) [inline SwrveConversationSDK/src/main/java/com/swrve/sdk/conversations/ui/ConversationFragment.java]: ~ ~ Double ;

**Comment by Sergio-Mira** [comment]
🚀

File: SwrveConversationSDK/src/main/java/com/swrve/sdk/conversations/ui/ConversationRoundedLinearLayout.java (hunk 3 above)
Comment (dominicmarmionswrve) [inline SwrveConversationSDK/src/main/java/com/swrve/sdk/conversations/ui/ConversationRoundedLinearLayout.java]: Yup, will do

**Pull request merged**